ui/src/components/Publication/RelatedPublications/VotingArea/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import * as OutlineIcons from '@heroicons/react/24/outline';
+
+import * as api from '@/api';
+import * as Components from '@/components';
+import * as Config from '@/config';
+import * as I from '@/interfaces';
+import * as Stores from '@/stores';
+import * as Types from '@/types';
+
+type Props = {
+    suggestedFromPublication: {
+        id: string;
+        title: string;
+    };
+    crosslink: I.Crosslink;
+    vote: boolean | null;
+};
+
+const VotingArea: React.FC<Props> = (props): React.ReactElement => {
+    const user = Stores.useAuthStore((state: Types.AuthStoreType) => state.user);
+    const [userHasUpvoted, setUserHasUpvoted] = useState<boolean>(props.vote === true);
```

Comment:
The way I get the vote, it can be true (voted up), false (voted down) or null (didn't vote). It would be simpler for the state's initial values if it was only true or false, but because of that I was more explicit with the === true etc.

Reply:
:+1: `!!props.vote` would help there, but if going the explicit route that makes sense